server/handler.go

```diff
@@ -486,28 +487,32 @@ func (h *Handler) AddTransferLeaderOperator(regionID uint64, storeID uint64) err
 }
 
 // AddTransferRegionOperator adds an operator to transfer region to the stores.
-func (h *Handler) AddTransferRegionOperator(regionID uint64, storeIDs map[uint64]struct{}) error {
+func (h *Handler) AddTransferRegionOperator(regionID uint64, storeIDsAndRoles map[uint64]placement.PeerRoleType) error {
 	c, err := h.GetRaftCluster()
 	if err != nil {
 		return err
 	}
 
 	if c.GetOpts().IsPlacementRulesEnabled() {
-		// Cannot determine role when placement rules enabled. Not supported now.
-		return errors.New("transfer region is not supported when placement rules enabled")
+		for _, peerRole := range storeIDsAndRoles {
+			if peerRole == "" {
+				// Cannot omit role when placement rules enabled.
+				return errors.New("peer role must be provided when placement rules enabled")
+			}
+		}
 	}
 
 	region := c.GetRegion(regionID)
 	if region == nil {
 		return ErrRegionNotFound(regionID)
 	}
 
-	if len(storeIDs) > c.GetOpts().GetMaxReplicas() {
-		return errors.Errorf("the number of stores is %v, beyond the max replicas", len(storeIDs))
+	if len(storeIDsAndRoles) > c.GetOpts().GetMaxReplicas() {
+		return errors.Errorf("the number of stores is %v, beyond the max replicas", len(storeIDsAndRoles))
```

Comment:
When placement rules enabled, we cannot use `MaxReplicas` to determine max replicas for a region. I think we can remove this check directly.

Reply:
I added a condition check here. Only when placement rules disabled will the check be performed now, or should I remove the check completely?